install-template.sh

```diff
@@ -625,7 +626,7 @@ install_components() {
 
 		    maybe_backup_path "$_file_install_path"
 
-		    if echo "$_file" | grep "^bin/" > /dev/null || test -x "$_src_dir/$_component/$_file"
+		    if test -z "${_file##bin/*}" || test -x "$_src_dir/$_component/$_file"
```

Comment:
`test -z "${_file##bin/*}"` unintentionally accepts the empty string; `[ "${_file#bin/}" != "$_file" ]` would be a slightly more faithful translation. (Presumably this doesn’t really matter.)

Or there’s `[[ "$_file" = bin/* ]]` since this is `#!/bin/bash`, although the script seems to otherwise avoid bashisms so I guess that avoidance should be preserved.

Reply:
`${_file}` is never empty here because of https://github.com/rust-lang/rust-installer/pull/106/files#diff-ec68db39ae4ea5bfe559a47a8a880d5d383dfa81ba067652d2adfcc3c0cd2a17R571 .
So,
1. `if test -z "${_file##bin/*}" || ...`
2. `if [ "${_file#bin/*}" != "${_file}" ] || ...`
3. `if case "${_file}" in bin/*) ;; *) false ; esac || ...` (POSIX-y `[[ "${_file}" = bin/*`)
should all be equivalent here.
If someone has a strong preference for any of those options (I don't), then I'm happy to change it accordingly!